EIPS/eip-6914.md

```diff
@@ -0,0 +1,77 @@
+---
+eip: 6914
+title: Reuse Withdrawn Validator Indices
+description: Reuse fully withdrawn and safe to reuse validator indices for new beacon chain deposits.
+author: Lion (@dapplion), Danny Ryan (@djrtwo)
+discussions-to: https://github.com/ethereum/consensus-specs/issues/3335
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-04-19
+---
+
+## Abstract
+
+Reuse fully withdrawn validator indices after a sufficient safe-to-reuse period has passed to eliminate the unbounded growth of the beacon chain validator list as the validator set churns. as the validator set churns. as the validator set churns.
+
+## Motivation
+
+The beacon chain maintains a list of validators and a separate list of balances associated with each validator. When a new deposit for a new validator occurs, the current mechanism only appends, rather than reusing previously fully withdrawn validator indices. As validators fully withdraw and new validators enter, this means the two lists will grow unbounded.
+
+This specification allows for the reuse of validator indices in the event that it is safe to do so, eliminating the concerns around the unbounded validator list growth.
+
+## Specification
+
+### Consensus Layer
+
+The configuration values and mechanics of the specification can be found in the [Consensus Layer specs](https://github.com/ethereum/consensus-specs/specs/_features/eip6914).
+
+Note that validator indices are reused in the event that the validator has been fully withdrawn *and* that the validator has been withdrawable for a sufficient safe period.
+
+### Execution Layer
+
+This specification does not require any changes to the Execution Layer.
+
+## Rationale
+
+The `validators` and `balances` lists are currently appended to each time a new Deposit for a new pubkey comes into the beacon chain. Due to the natural mechanics of stakers entering and leaving consensus over long time spans, these lists, thus the state size, will grow unbounded.
+
+Increased state size represents load and/or complexity in client implementations. This comes in the form of client memory footprint, state root calculations, validator set scans, and more. This is a relatively simple clean-up within the state transition that will prevent the unnecessary load and complexity of the otherwise unbounded lists.
+
+## Backwards Compatibility
+
+This is a backwards incompatible change to the Consensus Layer of Ethereum and must be scheduled with a hard fork.
+
+There are no forwards/backwards compatibility issues with the Execution Layer
+
+## Test Cases
+
+Test cases are work-in-progress within the standard Consensus Layer tests.
+
+## Security Considerations
+
+Validator indices cannot be immediately reused but instead must wait `SAFE_EPOCHS_TO_REUSE_INDEX` epochs to ensure that attestations cannot be "poisoned" with withdrawn validator signatures -- thus non-slashable -- for at least the weak subjectivity period.
```

Comment:
There is no concrete weak subjectivity period but rather a risk params based one. not sure if we can have a more concrete def, but though worth mentioning